src/dmd/lexer.d

```diff
@@ -2769,17 +2774,28 @@ class Lexer
     }
 
     /*********************************************
-     * parse:
-     *      #line linnum [filespec]
-     * also allow __LINE__ for linnum, and __FILE__ for filespec
+     * Parse line/file preprocessor directive:
+     *    #line linnum [filespec]
+     * Allow __LINE__ for linnum, and __FILE__ for filespec.
+     * Accept linemarker format:
+     *    # linnum [filespec] {flags}
+     * There can be zero or more flags, which are one of the digits 1..4, and
+     * must be in ascending order. The flags are ignored.
+     * Params:
+     *  tok = token we're on, which is linnum of linemarker
+     *  linemarker = true if line marker format and lexer is on linnum
+     * References:
+     *  linemarker https://gcc.gnu.org/onlinedocs/gcc-3.0.2/cpp_9.html
```

Comment:
Link to more recent documentation.

https://gcc.gnu.org/onlinedocs/gcc-11.1.0/cpp/Preprocessor-Output.html

GCC 3.0.2 is almost 20 years old!

Reply:
The link I used is the top google search link to it.

Reply:
There's been a few changes to the documentation since then though - mostly for clarity.